Kills research scanner toggle, moves functionality to examine_more. Improves research scanner code and fixes a modsuit bug.

VladinXXV opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This PR removes the Research Scanner Toggle action button. The functionality it used to enable has been moved to `examine_more` for everything EXCEPT seeing extra details about a Blob. It also turns `research_scanner` into a trait, which should make the code a little nicer.  While testing this PR, my debug instance paused on a runtime due to an existing bug with the MOD advanced reagent scanner module, which was registering a signal twice, instead of registering it and unregistering it. So I fixed that, too.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The Research Scanner function of science goggles typically involves a very small amount of examine text, and has relatively niche uses. The toggle button has always seemed very unneccesary, and frankly it drove me insane. The uses for the function are pretty niche, and aren't disruptive to have constantly active. That said, making it a part of `examine_more` instead of just `examine` makes sure you aren't flooding your chat with a list of parts when you don't want to.

I don't understand EXACTLY why making `research_scanner` a trait is better, but if you look at the current code used in the action I'm removing here, it might be obvious? LemonInTheDark seemed to think it was an improvement.

And last but not least, bugs are bad, so fixing a bug is good!
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Vladoricious
qol: The research scanner in science goggles and similar eyewear no longer requires you to press a button to turn it on! Instead, simply closely examine your target.
fix: The MOD advanced reagent scanner module no longer detects explosions despite being disabled.
code: Research scanner code cleanup.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

VladinXXV committed:
FINALLY. I'VE KILLED IT. I CAN LIVE MY LIFE NOW.

I hate the fucking Toggle Research Scanner action button so god damn much. Why the fuck would I ever not want this to be on? Why do you think I'm wearing the fucking goggles? That stupid button is so annoying to use. Even if I'm NOT using the research scanner aspect of the goggles, that little shit floats there, taking up space on my screen, taunting me.

ArcaneDefence commented:
If you're removing the toggle you should remove the var from mob too.
Lord knows the doc is useless for it anyway.
`	/// I have no idea tbh
	var/research_scanner = FALSE`

VladinXXV commented:
> If you're removing the toggle you should remove the var from mob too. Lord knows the doc is useless for it anyway. ` /// I have no idea tbh var/research_scanner = FALSE`

I saw that but I wasn't 100% sure if it was needed. I'll try removing it, re-test things, and if it works I'll commit the change.

VladinXXV committed:
ArcaneDefense suggestion

"If you're removing the toggle you should remove the var from mob too.
Lord knows the doc is useless for it anyway."

Mothblocks commented on code/game/objects/items.dm:
```diff
@@ -342,10 +342,12 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/e
+	if(!HAS_TRAIT(user, TRAIT_RESEARCH_SCANNER))
+		return
 	/// Research prospects, including boostable nodes and point values. Deliver to a console to know whether the boosts have already been used.
 	var/list/research_msg = list("<font color='purple'>Research prospects:</font> ")
```
Can you make this its own proc that is called if they *do* have the trait? It would help make this code a lot easier to read.

VladinXXV replied:
I thiiink so? Check the latest version and tell me if that's what you were looking for.

VladinXXV committed:
I think this is what Mothblocks wanted?

send help the headmins are holding me hosta- GURK
GAK
GUH
*deathgasp

Fikou commented on code/game/objects/items.dm:
```diff
@@ -342,10 +342,15 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/e
+/obj/item/examine_more(mob/user)
+	. = ..()
+	if(HAS_TRAIT(user, TRAIT_RESEARCH_SCANNER))
+		research_scan(user)
+		. += research_scan()
```
```suggestion
		. += research_scan(user)
```

VladinXXV committed:
this is probably a good thing

dont ask me how

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>
